external/dhcpc/dhcpc.c

```diff
@@ -674,9 +688,9 @@ static int dhcpc_request(void *handle, struct dhcpc_state *presult)
     ip_addr_t dns_addr;
     IP_SET_TYPE_VAL(dns_addr, IPADDR_TYPE_V4);
 #ifdef CONFIG_NET_IPv6
-    dns_addr.u_addr.ip4.addr = presult->dnsaddr.s_addr;
+    dns_addr.u_addr.ip4.addr = presult->dnsaddr[0].s_addr;
 #else
-    dns_addr.addr = presult->dnsaddr.s_addr;
+    dns_addr.addr = presult->dnsaddr[0].s_addr;
```

Comment:
you have to set the number of  registered dns devices.
please refer req.num_dns structure member.

Reply:
Done. please check now.